src/PharoLauncher-CLI-Tests/ConsoleListFormatterTest.class.st

```diff
@@ -162,64 +181,58 @@ ConsoleListFormatterTest >> testPrintLineSized [
 ]
 
 { #category : #test }
-ConsoleListFormatterTest >> testPrintList [
+ConsoleListFormatterTest >> testPrintListAsSton [
 
-	| stonString tableModeString |
-	formatter attributeLabels: #('Value' 'Squared' 'Raised to 3').
-	formatter attributeValueBlocks: { [:a |a asWords  ]. [:a |a squared asWords  ]. [:a |(a **3) asWords  ] }.
-	formatter domainObjects: #(1 2 3).
-	
-	"test if table is printed, if ston mode is disabled"
-	formatter printList.
-	tableModeString  := formatter outStream contents.
-	command resetOutStream.
-	formatter printListAttributes.
-	self assert: formatter outStream contents equals: tableModeString.
+	| domainObjects deserialized |
+	domainObjects := { PhLImageMock atRandom. PhLImageMock atRandom. PhLImageMock atRandom }.
+	formatter domainObjects: domainObjects.
 
-	"test for ston output flag"
-	command resetOutStream.
-	formatter outputAsSton: true.
-	formatter printList.
-	stonString  := formatter outStream contents.
-	command resetOutStream.
 	formatter printListAsSton.
-	self assert: formatter outStream contents equals: stonString.
+	deserialized := STON fromString: formatter outStream contents.
+	
+	self assert: deserialized size equals: 3.
+	self assert: deserialized first class equals: PhLImageMock.
 ]
 
 { #category : #test }
-ConsoleListFormatterTest >> testPrintListAsSton [
-
-	formatter domainObjects: {PhLImagesMock atRandom. PhLImagesMock atRandom. PhLImagesMock atRandom.}.
-	formatter printListAsSton.
+ConsoleListFormatterTest >> testPrintListShouldPrintAttributesTableWhenStonAndRowModesDisabled [
+	| tableModeString |
+	formatter attributeLabels: #('Value' 'Squared' 'Raised to 3').
+	formatter
+		attributeValueBlocks:
+			{[ :a | a asWords ].
+			[ :a | a squared asWords ].
+			[ :a | (a ** 3) asWords ]};
+		domainObjects: #(1 2 3);
+		printAttributeTable.
+	tableModeString := formatter outStream contents.
+	command resetOutStream.
 
-	self assert: (formatter outStream contents includesSubstring: 'PhLImagesMock').
+	formatter printList.
+	
+	self assert: formatter outStream contents equals: tableModeString
 ]
 
 { #category : #test }
-ConsoleListFormatterTest >> testPrintListAttributes [
+ConsoleListFormatterTest >> testPrintListShouldPrintAttributesTableWhenStonDisabledAndRowModeEnabled [
 
-		| rowModeString tableModeString   |
-	formatter attributeLabels: #('Value' 'Squared' 'Raised to 3').
-	formatter attributeValueBlocks: { [:a |a asWords  ]. [:a |a squared asWords  ]. [:a |(a **3) asWords  ] }.
-	formatter domainObjects: #(1 2 3).
+	| rowModeString |
 	
-	"test if row mode works when using printListAttributes"
-	formatter rowMode: true.
-	formatter printListAttributes.
+	formatter attributeLabels: #('Value' 'Squared' 'Raised to 3').
+	formatter
+		attributeValueBlocks:
+			{[ :a | a asWords ].
+			[ :a | a squared asWords ].
+			[ :a | (a ** 3) asWords ]};
+		domainObjects: #(1 2 3);
+		rowMode: true;
+		printOneAttrPerRow.
```

Comment:
This test was originally testing, if printListAttributes method takes into account rowMode flag, now it calls a different method. Maybe this should still test printListAttributes, but one test with row mode enabled and another with rowMode disablem.

Reply:
Only testing `printListAttributes` is not really useful. See the implementation of `printListAttributes`:
```smalltalk
printListAttributes
	self domainObjects isEmptyOrNil
		ifTrue: [ ^ self ].
		
	self rowMode
		ifTrue: [ self printOneAttrPerRow ]
		ifFalse: [ self printAttributeTable ]
```
It is better to test directly printList. There were 2 tests testing the same thing.

Reply:
Ok, got it.